### PR TITLE
llvm: PtrToIntOp operation

### PR DESCRIPTION
**Pull request by AntonLydike**

### CODE REVIEW

File: xdsl/dialects/llvm.py [hunk 1]
```diff
@@ -144,6 +144,19 @@ def get(cls,
...
+@irdl_op_definition
+class PtrToIntOp(Operation):
+    name = "llvm.ptrtoint"
+
+    input: Annotated[Operand, LLVMPointerType]
+
+    output: Annotated[OpResult, IntegerType]
+
+    @classmethod
+    def get(cls, arg: SSAValue | Operation, int_type: Attribute = i64):
```
Comment (webmiche) [inline xdsl/dialects/llvm.py:156]: --> cls thing :wink:

**Review by webmiche — APPROVED** [review]
LGTM, modulo the classic comment :wink:

File: xdsl/dialects/llvm.py (hunk 1 above)
Comment (AntonLydike) [reply xdsl/dialects/llvm.py]: Fixed!

File: xdsl/dialects/llvm.py [hunk 2]
```diff
@@ -105,9 +105,8 @@ class AllocaOp(Operation):
...
     res: OpResult
 
-    @classmethod
-    def get(cls,
-            size: SSAValue | Operation,
+    @staticmethod
+    def get(size: SSAValue | Operation,
             elem_type: Attribute,
             alignment: int = 32,
             as_untyped_ptr: bool = False):
```
Comment (superlopuh) [inline xdsl/dialects/llvm.py:112]:
```suggestion
            as_untyped_ptr: bool = False) -> AllocaOp:
```

File: xdsl/dialects/llvm.py [hunk 3]
```diff
@@ -133,15 +132,26 @@ class IntToPtrOp(Operation):
 
     output: Annotated[OpResult, LLVMPointerType]
 
-    @classmethod
-    def get(cls,
-            input: SSAValue | Operation,
-            ptr_type: Attribute | None = None):
+    @staticmethod
+    def get(input: SSAValue | Operation, ptr_type: Attribute | None = None):
```
Comment (superlopuh) [inline xdsl/dialects/llvm.py:136]:
```suggestion
    def get(input: SSAValue | Operation, ptr_type: Attribute | None = None) -> IntToPtrOp:
```

File: xdsl/dialects/llvm.py [hunk 4]
```diff
@@ -133,15 +132,26 @@ class IntToPtrOp(Operation):
...
+@irdl_op_definition
+class PtrToIntOp(Operation):
+    name = "llvm.ptrtoint"
+
+    input: Annotated[Operand, LLVMPointerType]
+
+    output: Annotated[OpResult, IntegerType]
+
+    @staticmethod
+    def get(arg: SSAValue | Operation, int_type: Attribute = i64):
```
Comment (superlopuh) [inline xdsl/dialects/llvm.py:153]:
```suggestion
    def get(arg: SSAValue | Operation, int_type: Attribute = i64) -> PtrToIntOp:
```

**Review by superlopuh — APPROVED** [review]
LGTM
Added some suggestions for explicit return types, I kinda like them even if they're optional

File: xdsl/dialects/llvm.py (hunk 4 above)
Comment (AntonLydike) [reply xdsl/dialects/llvm.py:153]: Not sure on these, makes copy-pasting more difficult (don't know if that's a bad thing)

Comment (webmiche) [reply xdsl/dialects/llvm.py:153]: Copy-pasting of what? :sweat_smile: 
Of the code itself?

Comment (AntonLydike) [reply xdsl/dialects/llvm.py:153]: Well, if you add a new `get` that's quite similar to an older one, you'd have to replace one more occurance of `OldType` with `NewType`^^

**Pull request merged**